delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/routes/FilesRoutes.scala

```diff
@@ -184,13 +184,17 @@ final class FilesRoutes(
     }
 
   def fetch(id: IdSegmentRef, ref: ProjectRef)(implicit caller: Caller): Route =
-    headerValueByType(Accept) {
+    (headerValueByType(Accept) & varyAcceptHeaders) {
       case accept if accept.mediaRanges.exists(metadataMediaRanges.contains) =>
-        emit(fetchMetadata(id, ref).rejectOn[FileNotFound])
+        emit(fetchMetadata(id, ref).rejectWhen(unauthorizedOrNotFound))
       case _                                                                 =>
-        emit(files.fetchContent(id, ref).rejectOn[FileNotFound])
+        emit(files.fetchContent(id, ref).rejectWhen(unauthorizedOrNotFound))
     }
 
+  private val unauthorizedOrNotFound: PartialFunction[FileRejection, Boolean] = {
+    case _: AuthorizationFailed | _: FileNotFound => true
+  }
```

Comment:
It's necessary to do this as in the ResourcesRoutes. Without the rejection on AuthorizationFailed it will inject the Vary header into the errors.

Reply:
That's better yes